controllers/etcd_controller.go

```diff
@@ -1045,3 +1055,24 @@ func (r *EtcdReconciler) claimConfigMaps(etcd *druidv1alpha1.Etcd, selector labe
 	cm := NewEtcdDruidRefManager(r, etcd, selector, etcdGVK, canAdoptFunc)
 	return cm.ClaimConfigMaps(ss)
 }
+
+// SetupWithManager sets up manager with a new controller and r as the reconcile.Reconciler
+func (r *EtcdReconciler) SetupWithManager(mgr ctrl.Manager, workers int, ignoreOperationAnnotation bool) error {
+	predicates := []predicate.Predicate{
+		druidpredicates.GenerationChangedPredicate{},
+		druidpredicates.LastOperationNotSuccessful(),
+	}
+	builder := ctrl.NewControllerManagedBy(mgr).WithOptions(controller.Options{
+		MaxConcurrentReconciles: workers,
+	})
+	if !ignoreOperationAnnotation {
+		predicates = append(predicates, druidpredicates.HasOperationAnnotation())
+	}
+	builder = builder.WithEventFilter(druidpredicates.Or(predicates...))
```

Comment:
@timuthy - what about the case if the deletionTimestamp is set? Is a predicate needed for it?

Reply:
If I don't add a predicate for deletionTimestamp, it will still reconcile. Do I need to add it explicitly?

Reply:
OK, it should actually reconcile if the deletion timestamp is set. How is this ensured?

Reply:
OK. we just figured out that the generation is increased when the  `Etcd` resource gets deleted. Hence, nothing to do.